src/main/java/org/jboss/logmanager/LogContext.java

```diff
@@ -529,20 +495,7 @@ public void close() throws Exception {
             for (AutoCloseable handler : closeHandlers) {
                 handler.close();
             }
-            synchronized (this) {
-                attachmentKey1 = null;
-                attachmentKey2 = null;
-                final var value1 = attachmentValue1;
-                attachmentValue1 = null;
-                if (value1 instanceof AutoCloseable) {
-                    ((AutoCloseable) value1).close();
-                }
-                final var value2 = attachmentValue2;
-                attachmentValue2 = null;
-                if (value2 instanceof AutoCloseable) {
-                    ((AutoCloseable) value2).close();
-                }
-            }
+            attachmentHandle.setVolatile(this, Map.of());
```

Comment:
Do we want to keep the above `AutoCloseable` checks? If so, this needs to be `getAndSetVolatile` and the return value must be iterated for closeable values.